tests/std/tests/P0896R4_ranges_alg_move/test.cpp

```diff
@@ -18,66 +20,63 @@ struct int_wrapper {
-    using std::same_as;
+// Validate that move_result aliases in_out_result
+STATIC_ASSERT(same_as<ranges::move_result<int, double>, ranges::in_out_result<int, double>>);
 
-    // Validate that move_result aliases in_out_result
-    STATIC_ASSERT(same_as<move_result<int, double>, ranges::in_out_result<int, double>>);
+// Validate dangling story
+STATIC_ASSERT(same_as<decltype(ranges::move(borrowed<false>{}, static_cast<int*>(nullptr))),
+    ranges::move_result<ranges::dangling, int*>>);
+STATIC_ASSERT(
+    same_as<decltype(ranges::move(borrowed<true>{}, static_cast<int*>(nullptr))), ranges::move_result<int*, int*>>);
 
-    // Validate dangling story
-    STATIC_ASSERT(
-        same_as<decltype(move(borrowed<false>{}, static_cast<int*>(nullptr))), move_result<ranges::dangling, int*>>);
-    STATIC_ASSERT(same_as<decltype(move(borrowed<true>{}, static_cast<int*>(nullptr))), move_result<int*, int*>>);
 
-    int const input[] = {13, 53, 12435};
-    {
-        int output[] = {-2, -2, -2};
-        auto result  = move(basic_borrowed_range{input}, basic_borrowed_range{output}.begin());
-        STATIC_ASSERT(same_as<decltype(result),
-            move_result<iterator_t<basic_borrowed_range<int const>>, iterator_t<basic_borrowed_range<int>>>>);
-        assert(result.in == basic_borrowed_range{input}.end());
-        assert(result.out == basic_borrowed_range{output}.end());
-        assert(ranges::equal(output, input));
-    }
-    {
-        int output[] = {-2, -2, -2};
-        basic_borrowed_range wrapped_input{input};
-        auto result = move(wrapped_input.begin(), wrapped_input.end(), basic_borrowed_range{output}.begin());
-        STATIC_ASSERT(same_as<decltype(result),
-            move_result<iterator_t<basic_borrowed_range<int const>>, iterator_t<basic_borrowed_range<int>>>>);
-        assert(result.in == wrapped_input.end());
-        assert(result.out == basic_borrowed_range{output}.end());
-        assert(ranges::equal(output, input));
-    }
-    {
-        int_wrapper input1[3]        = {13, 55, 1234};
-        int const expected_output[3] = {13, 55, 1234};
-        int_wrapper actual_output[3] = {-2, -2, -2};
-        basic_borrowed_range wrapped_input{input1};
-        auto result = move(wrapped_input.begin(), wrapped_input.end(), basic_borrowed_range{actual_output}.begin());
-        assert(result.in == wrapped_input.end());
-        assert(result.out == basic_borrowed_range{actual_output}.end());
-        for (int i = 0; i < 3; ++i) {
-            assert(input1[i].val == -1);
-            assert(actual_output[i].val == expected_output[i]);
+struct instantiator {
+    static constexpr int input[3]                   = {13, 53, 12435};
+    static constexpr int_wrapper input1[3]          = {13, 55, 12345};
+    static constexpr int_wrapper expected_output[3] = {13, 55, 12345};
+    static constexpr int_wrapper expected_input[3]  = {-1, -1, -1};
+
+    template <class Read, class Write>

```

Comment:
```suggestion
    template <ranges::input_range Read, indirectly_writable<ranges::range_reference_t<Read>> Write>
```

Reply:
Done

Reply:
I think this shhould be `ranges::range_rvalue_reference_t
`